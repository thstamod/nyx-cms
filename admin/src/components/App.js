import React, { useEffect } from 'react';
import { BrowserRouter, Route, Switch, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import propTypes from 'prop-types';

import styled, { ThemeProvider } from 'styled-components';
import { Container } from 'react-bootstrap';
import AuthPage from '../pages/AuthPage/AuthPage';
import UsersPage from '../pages/UserPage/UersPage';
import ContentPage from '../pages/ContentPage/ContentPage';
import SettingsPage from '../pages/SettingsPage/SettingsPage';
import NotFound from '../pages/NotFound/NotFound';
import WithAuth from '../containers/withAuth';
import store from '../redux/store';
import { setSessionStorage } from '../utils/handleSessionStorage';
import GlobalStyles from '../theme/globalStyle';
import theme from '../theme/index';
import MainNavigation from './Navigation/MainNavigation';

const FullContainer = styled(Container)`
  padding-right: 0;
  padding-left: 0;
  height: 100%;
  overflow: hidden;
  padding-top: 56px;
`;

const App = ({ isLoggedIn }) => {
  useEffect(() => {
    window.addEventListener('beforeunload', (ev) => {
      ev.preventDefault();
      setSessionStorage(store);
    });
  }, []);
  return (
    <BrowserRouter>
      <GlobalStyles />
      <ThemeProvider theme={theme}>
        {isLoggedIn && <MainNavigation />}
        <React.Fragment>
          <FullContainer fluid>
            <Switch>
              <Redirect from="/" to="/auth" exact />
              <Route path="/auth" component={AuthPage} />
              <WithAuth path="/content">
                <ContentPage />
              </WithAuth>
              <WithAuth path="/settings">
                <SettingsPage />
              </WithAuth>
              <WithAuth path="/users">
                <UsersPage />
              </WithAuth>
              <Route path="*" component={NotFound} />
            </Switch>
          </FullContainer>
        </React.Fragment>
      </ThemeProvider>
    </BrowserRouter>
  );
};

App.propTypes = {
  isLoggedIn: propTypes.bool,
};

const mapStateToProps = (state) => {
  const { user } = state;
  return { isLoggedIn: user.isLoggedIn };
};

export default connect(mapStateToProps)(App);
