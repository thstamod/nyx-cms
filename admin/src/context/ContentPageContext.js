import React, { createContext, useContext } from 'react';
import {
  initialState,
  contentPageReducer,
} from '../state/reducers/contentPageReducer';
import useEnhancedReducer from '../hooks/useEnhancedReducer';
import logger from '../middlewares/customLogger';

export const ContentPageContext = createContext(null);

export function useContentPageState() {
  return useContext(ContentPageContext);
}

export function ContentPageProvider({ children }) {
  const middlewares = [logger];
  const store = useEnhancedReducer(
    contentPageReducer,
    initialState,
    middlewares
  );

  return (
    <ContentPageContext.Provider value={store}>
      {children}
    </ContentPageContext.Provider>
  );
}
