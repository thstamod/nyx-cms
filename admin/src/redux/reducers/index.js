import { combineReducers } from "redux";
import loginReducer from './loginReduser'

export default combineReducers({ loginReducer });
