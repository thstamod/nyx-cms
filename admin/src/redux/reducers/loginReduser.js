import { LOGIN } from "../actions/actionTypes";


const initialState = {
  token: ''
};

export default function (state = initialState, action) {
  switch (action.type) {
    case LOGIN: {
      const { token } = action.payload;
      return { token };
    }
    default:
      return state;
  }
}
