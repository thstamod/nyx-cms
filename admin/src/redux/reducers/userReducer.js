import { LOGIN, LOGOUT } from '../actions/actionTypes';

export const initialState = {
  isLoggedIn: false,
  token: '',
  tokenExpiration: null,
};

export function userReducer(state = initialState, action) {
  switch (action.type) {
    case LOGIN: {
      const { token, tokenExpiration } = action.payload;
      return { ...state, isLoggedIn: true, token, tokenExpiration };
    }
    case LOGOUT: {
      return { ...initialState };
    }
    default:
      return state;
  }
}
