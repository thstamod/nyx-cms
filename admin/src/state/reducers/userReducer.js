import { LOGIN, LOGOUT, GET_FROM_SESSION_STATE } from '../actions/actionTypes';
import { setSessionStorage } from '../../utils/handleSessionStorage';

export const initialState = {
  isLoggedIn: false,
  token: '',
  tokenExpiration: null,
};

export function userReducer(state = initialState, action) {
  switch (action.type) {
    case LOGIN: {
      const { token, tokenExpiration } = action.payload;
      setSessionStorage('user', {
        ...state,
        isLoggedIn: true,
        token,
        tokenExpiration,
      });
      return { ...state, isLoggedIn: true, token, tokenExpiration };
    }
    case LOGOUT: {
      return { ...initialState };
    }
    case GET_FROM_SESSION_STATE: {
      return { ...state, ...action.payload };
    }
    default:
      return state;
  }
}
