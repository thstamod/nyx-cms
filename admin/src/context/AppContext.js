import React, { createContext, useContext } from 'react';
import { userReducer, initialState } from '../redux/reducers/userReducer';
import useEnhancedReducer from '../hooks/useEnhancedReducer';
import logger from '../middlewares/customLogger';

export const AppContext = createContext();

export function useAppState() {
  return useContext(AppContext);
}

export function AppStateProvider({ children }) {
  const middlewares = [logger];
  const store = useEnhancedReducer(userReducer, initialState, middlewares);

  return <AppContext.Provider value={store}>{children}</AppContext.Provider>;
}
