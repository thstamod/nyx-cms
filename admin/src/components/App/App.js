import React from 'react';
import { BrowserRouter, Route, Switch, Redirect } from 'react-router-dom';
import AuthPage from '../../pages/AuthPage/AuthPage';
import UsersPage from '../../pages/UserPage/UserPage';
import ContentPage from '../../pages/ContentPage/ContentPage';
import SettingsPage from '../../pages/SettingsPage/SettingsPage';
import NotFound from '../../pages/NotFound/NotFound';
import AuthRoute from '../AuthRoute/AuthRoute';
import GlobalStyles from '../../theme/globalStyle';
import MainNavigation from '../Navigation/MainNavigation/MainNavigation';
import { useAppState } from '../../context/AppContext';

const App = () => {
  const [{ isLoggedIn }] = useAppState();

  return (
    <BrowserRouter>
      <GlobalStyles />
      {isLoggedIn && <MainNavigation />}
      <React.Fragment>
        <Switch>
          {!isLoggedIn ? (
            <Redirect from="/" to="/auth" exact />
          ) : (<Redirect from="/" to="/content" exact />)}
          {!isLoggedIn && <Route path="/auth" component={AuthPage} />}
          {isLoggedIn && (
            <AuthRoute path="/content" component={ContentPage} />
          )}
          {isLoggedIn && (
            <AuthRoute path="/settings" component={SettingsPage} />
          )}
          {isLoggedIn && <AuthRoute path="/users" component={UsersPage} />}
          <Route path="*" component={NotFound} />
        </Switch>
      </React.Fragment>
    </BrowserRouter>
  );
};

export default App;
