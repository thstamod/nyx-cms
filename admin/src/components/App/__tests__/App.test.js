import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, waitFor, cleanup } from '@testing-library/react';
import { AppContext } from '../../../context/AppContext';
import App from '../App';
import { initialState } from '../../../state/reducers/userReducer';

afterEach(cleanup);

function renderWithProvider(store) {
  return render(
    <AppContext.Provider value={[store]}>
      <App />
    </AppContext.Provider>
  );
}

test('App renders', () => {
  const { asFragment } = renderWithProvider(initialState);

  expect(asFragment()).toMatchSnapshot();
});

test('App initial state', async () => {
  const { getByText } = renderWithProvider(initialState);

  waitFor(() => expect(getByText(/^Submit/)).toBeInTheDocument());
});

test('App logged in state', async () => {
  const state = {
    isLoggedIn: true,
    token:
      'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOiI1ZTcxNDczZjE4MTIxYzJmZjI0YzQwZWYiLCJlbWFpbCI6InRlc3RAbWFpbC5jb20iLCJpYXQiOjE1ODc4MDkxOTUsImV4cCI6MTU4NzgxMjc5NX0.2Rnz5e8-YwcvNrqcS1C8WGUddjMjygn9yJutfWdNC_Q',
    tokenExpiration: 1587812795962,
  };

  const { getByTestId } = renderWithProvider(state);
  waitFor(() => expect(getByTestId('main-container')).toBeInTheDocument());
});
