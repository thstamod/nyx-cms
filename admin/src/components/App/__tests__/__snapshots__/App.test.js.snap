// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<DocumentFragment>
  <form>
    <div>
      <label
        for="email"
      >
        Email
      </label>
      <input
        id="email"
        type="email"
      />
      <label
        for="password"
      >
        Password
      </label>
      <input
        id="password"
        type="password"
      />
    </div>
    <button
      type="button"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
